lms/djangoapps/badges/tests/test_models.py

```diff
@@ -28,13 +28,6 @@
 from xmodule.modulestore.tests.factories import CourseFactory  # lint-amnesty, pylint: disable=wrong-import-order
 
 
-def get_image(name):
-    """
-    Get one of the test images from the test data directory.
-    """
-    return ImageFile(open(f'{TEST_DATA_DIR}/badges/{name}.png', mode='rb'))  # lint-amnesty, pylint: disable=bad-option-value, open-builtin
```

Comment:
I'd tried this approach at first but some tests expect the `ImageFile` object to be open for reading but with the suggested approach, the tests fail so we'll need to do the refactoring in either case. 
In this case, we would've needed to change the actual tests functionality so I opted for the current choice of refactoring instead.

Reply:
I think this would do what you want:
```python
@contextlib.contextmanager
def get_image(name):
    with open(f'{TEST_DATA_DIR}/badges/{name}.png', mode='rb') as fimage:
        yield ImageFile(fimage)

...

with get_image('good') as icon:
    CourseCompleteImageConfiguration(mode='test', icon=icon, default=True).save()
```

Reply:
Updated as per the suggestion.